include/shark/Data/SparseData.h

```diff
@@ -173,39 +164,45 @@ SHARK_EXPORT_SYMBOL void importSparseData(
 ///
 /// \param  dataset     Container storing the  data
 /// \param  stream      Output stream
-/// \param  dense       Flag for writing all components, including zeros
 /// \param  oneMinusOne Flag for applying the transformation y<-2y-1 to binary labels
 /// \param  sortLabels  Flag for sorting data points according to labels
 template<typename InputType>
-void exportSparseData(LabeledData<InputType, unsigned int> const& dataset, std::ostream& stream, bool dense=false, bool oneMinusOne = true, bool sortLabels = false)
+void exportSparseData(LabeledData<InputType, unsigned int> const& dataset, std::ostream& stream, bool oneMinusOne = true, bool sortLabels = false)
 {
 	std::size_t elements = dataset.numberOfElements();
 
 	size_t dim = inputDimension(dataset);
 	if(numberOfClasses(dataset)!=2) oneMinusOne = false;
 
-	std::vector<detail::LabelSortPair> L;
-	if (sortLabels) {
-		for(std::size_t i = 0; i < elements; i++) 
-			L.push_back(detail::LabelSortPair(dataset.element(i).label, i));
-		std::sort (L.begin(), L.end(), detail::cmpLabelSortPair);
+	std::vector< std::pair<std::size_t, std::size_t> > order;
```

Comment:
using KeyLabelPair would have been much(!) easier, there you get the op< for free